datastore_entity/src/lib.rs

```diff
@@ -141,6 +150,87 @@ where
     }
 }
 
+fn generate_empty_key(kind: String) -> Key {
+    Key {
+        partition_id: None,
+        path: Some(vec![PathElement {
+            id: None,
+            kind: Some(kind),
+            name: None,
+        }]),
+    }
+}
+
+pub fn commit_one<A>(
+    entity: DatastoreEntity,
+    kind: String,
+    token: A,
+    project_name: &String,
+) -> Result<DatastoreEntity, String>
+where
+    A: ::google_api_auth::GetAccessToken + 'static,
+{
+    let mut result_entity = entity.clone();
+    let client = Client::new(token);
+    let projects = client.projects();
+    let builder = projects.begin_transaction(
+        BeginTransactionRequest {
+            transaction_options: None,
+        },
+        project_name,
+    );
+    let begin_transaction: BeginTransactionResponse =
+        builder
+            .execute()
+            .map_err(|_e: google_datastore1::Error| -> String {
+                "Begin transaction request failed".to_string()
+            })?;
+    let is_insert = !entity.has_key();
+    let key = entity
+        .key()
+        .unwrap_or_else(|| -> Key { generate_empty_key(kind) });
+    let properties: DatastoreProperties = DatastoreProperties::from(entity).unwrap();
```

Comment:
Implement this instead?

```rust
impl TryFrom<DatastoreEntity> for Entity {
 ...
}
```

then you can do `entity.try_into().map_err(|_| "some err".to_string())?` instead?

Reply:
yes, sounds smart, that I will do!

Reply:
fixed in https://github.com/andreaspsson/datastorers/pull/16

Reply:
fixed in https://github.com/andreaspsson/datastorers/pull/16